ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -217,9 +217,9 @@ protected int getPartitions(final KafkaTopicClient kafkaTopicClient) {
   void validateKeyPresent(final SourceName sinkName, final Projection projection) {
 
     if (joinKey.isForeignKey()) {
-      final DataSource leftInputTable = getLeftmostSourceNode().getDataSource();
-      final SourceName leftInputTableName = leftInputTable.getName();
-      final List<Column> leftInputTableKeys = leftInputTable.getSchema().key();
+      final DataSourceNode leftInputTable = getLeftmostSourceNode();
+      final SourceName leftInputTableName = leftInputTable.getAlias();
```

Comment:
This is the actual fix: `getAlias` return the alias is there is one; otherwise the table name.